src/synapse-plugins/converter-plugin/converter-plugin-backend.vala

```diff
@@ -100,44 +102,24 @@ namespace Synapse {
             bool use_prefix = false;
 
             if (matched) {
-                // Parse input into a number and two unit match arrays
                 // Some abbreviations are ambiguous (used in >1 system) so get all possible matching units
-
-                string[] parts;
-                if (input.contains ("=>")) {
-                    parts = input.split ("=>", 2);
-                } else {
-                    parts = input.split ("->", 2);
-                }
-                var num_s = parts[0];
-                // Isolate leading number (First \0 truncates string)
-                num_s.canon ("1234567890.", '\0');
-                if (num_s.contains ("..")) { // Multiple decimal points are incorrectly parsed to double
-                    return results;
-                }
-
-                if (num_s.length > 0) { // If leading number omitted, assume it to be 1.0
-                    num = double.parse (num_s);
-                }
-
-                // Get user input units; cleanup excess whitespace. Note: more than two consecutive embedded spaces
-                // will prevent a match.
-                string unit1_s = (parts[0].slice (num_s.length, parts[0].length)).strip ().replace ("  ", " ");
-                string unit2_s = parts[1].strip ().replace ("  ", " ");
-
-                // Split each unit into prefix, base and dimension
-                get_prefix_and_dimension (unit1_s, out prefix1_s, out prefix1, out dimension1);
-                get_prefix_and_dimension (unit2_s, out prefix2_s, out prefix2, out dimension2);
-                debug ("unit1_s %s, unit2_s, %s, prefix1 %s, prefix2 %s, dimension1 %i, dimension2 %i",
-                       unit1_s, unit2_s, prefix1_s, prefix2_s, dimension1, dimension2
+                num = double.parse (match_info.fetch (1));
+                num = num == 0 ? 1.0 : num;
+                var unit1_s = match_info.fetch (2).strip ();
+                dimension1 = int.parse (match_info.fetch (3)).clamp (1, 3);
```

Comment:
Since you only match 2 or 3 as dimension why would you need to clamp it?

Reply:
I clamped it because if empty the dimension would otherwise be `0`.

Reply:
Tests fail if clamp () removed.

Reply:
Ahh, I get it now :+1: